### ISSUE TITLE
bug(Slider): Page Up/ Page Down buttons increase value but not by 10 steps

### ISSUE THREAD

**Issue by elenachiosa** [angular/components#27402]
### Is this a regression?

- [x] Yes, this behavior used to work in the previous version

### The previous version in which this bug was not present was

14.2.7

### Description

According to the documentation, the slider should increase the value by 10 steps when using Page Up/Page Down keys.
It instead increases/decreases by a different number of steps.
![image](https://github.com/angular/components/assets/17402311/ff4eced7-3134-4913-8a04-f086ebbc8f2c)
![slidermaterialissuepageupdown](https://github.com/angular/components/assets/17402311/38a787cd-dbe2-48b4-9847-4eed9b60eef3)

### Reproduction

StackBlitz link: 500 Internal server error for Stackblitz.... but the issue can be reproduced on the official Material demo
Steps to reproduce:
1. Go to: https://material.angular.io/components/slider/examples#slider-configurable
2. Type '9999' as the max value
3. Use the Page Up/Page Down keys to increase/decrease value

### Expected Behavior

The value is increased/decreased by 10 steps.

### Actual Behavior

The value is increased/decreased by 1000 steps for max value 9999, 
the value is increased/decreased by 100 steps for max value 1000,
not by 10 steps, like in the documentation.

The step is not set, so I assumed a default of 1 x 10 = 10 steps.

Using the arrows, the value is increased/decreased correctly, by 1 step.
Happens on both angular 15 and 16.

### Environment

- Angular: 15.2.9
- CDK/Material: 15.2.9
- Browser(s): Chrome
- Operating System (e.g. Windows, macOS, Ubuntu): Windows

**Comment by mmalerba** [comment]
I suspect this may be intentional and the documentation is just out of date (incrementing by 10 doesn't get you very far in a slider with 1k steps). The [WAI ARIA docs](https://www.w3.org/WAI/ARIA/apg/patterns/slider/) don't prescribe a specific increment for page-up/page-down, just that it should be more than regular up/down.

**Comment by wagnermaciel** [comment]
+1 to what @mmalerba said. This behavior was not added to the slider by us. It is the behavior of the hidden native `<input type="range"` that is being used under the hood

**Comment by lashagelashvili** [comment]
Was this issue fixed, and can't seem to reproduce it?

**Comment by lashagelashvili** [comment]
Was this issue fixed?  Can't seem to reproduce it.

**Comment by allexgut** [comment]
It makes sense that the amount increases with the max value - as @mmalerba said, 10 steps in a slider with 1k steps won't get you very far. Should then the docs be updated to be more descriptive?
